### ISSUE TITLE
Suggest the least powerful form of quotes applicable to the contents

### ISSUE THREAD

**Issue by l0b0** [koalaman/shellcheck#2594]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash
cat << EOF
Some static text
EOF
echo $'More static text'
```

#### Here's what shellcheck currently says:

> No issues detected!

#### Here's what I wanted or expected to see:

> Use single quoted here document start delimiter for static text.

```sh
cat << 'EOF'
Some static text
EOF
```

> Use single quoted string for static text

```sh
echo 'More static text'
```

[Rationale](https://paperless.blog/bash-lints-beyond-shellcheck): This is an example of the [rule of least power](https://en.wikipedia.org/w/index.php?title=Rule_of_least_power&oldid=1046246287). Using a less powerful language construct constrains the complexity of the contents of a string, lowering cognitive load.

This is analogous to [Pylint’s f-string without interpolation warning](https://pylint.pycqa.org/en/latest/messages/warning/f-string-without-interpolation.html).
